cli/command/image/build.go

```diff
@@ -226,6 +192,10 @@ func runBuild(dockerCli command.Cli, options buildOptions) error {
 		remote        string
 	)
 
+	if len(warnLegacyBuilder) > 0 && !options.quiet && dockerCli.ServerInfo().OSType != "windows" {
```

Comment:
Micro-nit: I generally prefer `warnLegacyBuilder != ""` over `len(warnLegacyBuilder) > 0` (I think it's slightly clearer we're checking a string (without having to look up what `warnLegacyBuilder` is.


```suggestion
	if warnLegacyBuilder != "" && !options.quiet && dockerCli.ServerInfo().OSType != "windows" {
```